### PR TITLE
New spring-web scenario

### PR DESCRIPTION
**Pull request by jsmrcka**
Based on https://github.com/eugenp/tutorials/tree/master/spring-boot-modules/spring-boot-bootstrap.

Adopting process tracked in https://github.com/jsmrcka/tutorials/tree/quarkus-spring-web.

Related to https://issues.redhat.com/browse/QUARKUS-185.

Unresolved issue: https://github.com/quarkusio/quarkus/issues/16321

### CODE REVIEW

File: 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/persistence/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+package com.baeldung.springbootbootstrap.persistence.model;
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/persistence/model/Book.java:1]: Rename the package name to `io.quarkus.qe.XX`

File: 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/persistence/repo/BookRepository.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.baeldung.springbootbootstrap.persistence.repo;
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/persistence/repo/BookRepository.java:1]: Rename the package name to `io.quarkus.qe.XX`

File: 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/BookController.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package com.baeldung.springbootbootstrap.web;
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/BookController.java:1]: Rename the package name to `io.quarkus.qe.XX`

File: 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/RestExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package com.baeldung.springbootbootstrap.web;
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/RestExceptionHandler.java:1]: Rename the package name to `io.quarkus.qe.XX`

File: 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/SimpleController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package com.baeldung.springbootbootstrap.web;
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/SimpleController.java:1]: Rename the package name to `io.quarkus.qe.XX`

File: 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/exception/BookIdMismatchException.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.baeldung.springbootbootstrap.web.exception;
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/exception/BookIdMismatchException.java:1]: Rename the package name to `io.quarkus.qe.XX`

File: 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/exception/BookNotFoundException.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package com.baeldung.springbootbootstrap.web.exception;
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/main/java/com/baeldung/springbootbootstrap/web/exception/BookNotFoundException.java:1]: Rename the package name to `io.quarkus.qe.XX`

File: 603-spring-web-smallrye-openapi/src/test/java/com/baeldung/springbootbootstrap/SpringBootBootstrapLiveTest.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+
+@QuarkusTest
+public class SpringBootBootstrapLiveTest {
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/test/java/com/baeldung/springbootbootstrap/SpringBootBootstrapLiveTest.java:21]: I didn't get the point of naming this test class `SpringBootBootstrapLiveTest`. What about BookResourceTest or BookRestResourceTest?

File: 603-spring-web-smallrye-openapi/src/test/java/com/baeldung/springbootbootstrap/HomePageTest.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+@QuarkusTest
+public class HomePageTest {
+    private static final String APP_NAME = "Bootstrap Spring Boot";
+
+    @Test
+    public void testHomePage() {
```
Comment (Sgitario) [inline 603-spring-web-smallrye-openapi/src/test/java/com/baeldung/springbootbootstrap/HomePageTest.java:16]: As far I understood, you're covering the qute extension here. What about renaming the test method to shouldQuteReplaceWelcomePharse or something similar?

**Review by Sgitario — CHANGES_REQUESTED** [review]
The package names should change from `com.baeldung` to `io.quarkus.qe`. Also, the package name `springbootbootstrap` is a bit verbose for me: what about simply `books`?

**Comment by Sgitario** [comment]
Depends on https://github.com/quarkus-qe/beefy-scenarios/pull/163

File: 603-spring-web-smallrye-openapi/src/test/java/com/baeldung/springbootbootstrap/SpringBootBootstrapLiveTest.java (hunk 8 above)
Comment (jsmrcka) [reply 603-spring-web-smallrye-openapi/src/test/java/com/baeldung/springbootbootstrap/SpringBootBootstrapLiveTest.java]: I went on with `BookResourceTest`.

**Comment by jsmrcka** [comment]
@Sgitario Thanks for the review, I updated all the naming as you suggested.

**Review by Sgitario — APPROVED** [review]

**Comment by rsvoboda** [comment]
@jsmrcka please rebase

**Pull request merged**